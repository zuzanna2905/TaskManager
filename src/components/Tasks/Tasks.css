::-webkit-scrollbar {
    width: 15px;
}

::-webkit-scrollbar-track {
    background: #c0b86e; 
}

::-webkit-scrollbar-thumb {
    background: rgb(129, 78, 20); 
}

::-webkit-scrollbar-thumb:hover {
    background: rgba(126, 83, 19, 0.692); 
}

::-webkit-scrollbar-button { 
    display: none; 
}