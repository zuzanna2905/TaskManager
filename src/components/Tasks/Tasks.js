import React from 'react'
import ListItem from '../ListItem/ListItem';
const url = 'http://localhost:3001/tasks';

class Tasks extends React.Component {
  listItems = () => {
    if(this.props.tasks){
      return this.props.tasks.map((task) => 
      <ListItem 
        key={task.id}
        value={task.task} 
        id={task.id}
        category={task.category}
        priority={task.priority}
        handleDelete={this.handleDelete} 
        handleDetails = {this.handleDetails}
        handleEdit = {this.handleEdit}     
      />);
    }
  }

  TaskList = () => this.listItems();

  handleDelete = (taskid) => {
    let deleteUrl = url + '/' + taskid;
    fetch(deleteUrl, {
      method: 'delete',
      headers: {'Content-Type' : 'application/json'},
    })
    .then(response => response.json())
    .then(
      this.props.loadData()
    )
  }

  render() {
    return (
      <div>
        <ul className="list">
          <this.TaskList />
        </ul>
      </div>
      )
  }
}

export default Tasks;