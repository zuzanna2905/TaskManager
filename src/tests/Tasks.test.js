import {shallow} from 'enzyme';
import React from 'react';
import Tasks from '../components/Tasks/Tasks';

const initialProps = {
    tasks : [
        {
            task: 'room cleaning',
            id: 1
        },
        {
            task: 'dish washing',
            id: 2
        }
    ]
}
const initialProps2 = {}

it('renders Tasks without crashing with data', () => {
    let wrapper = shallow(<Tasks {...initialProps}/>);
    expect(wrapper).toMatchSnapshot();
});

it('simulate listItems', () => {    
    let wrapper2 = shallow(<Tasks {...initialProps}/>);
    wrapper2.instance().listItems();
    wrapper2.instance().TaskList();
})

it('renders Tasks without crashing without data', () => {
    let wrapper3 = shallow(<Tasks {...initialProps2}/>);
    expect(wrapper3).toMatchSnapshot();
});