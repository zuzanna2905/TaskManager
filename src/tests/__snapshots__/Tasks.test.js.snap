// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tasks without crashing with data 1`] = `
<div
  className="mv4"
>
  <ul
    className="overflow-y-auto vh-50 list"
  >
    <Component />
  </ul>
</div>
`;

exports[`renders Tasks without crashing without data 1`] = `
<div
  className="mv4"
>
  <ul
    className="overflow-y-auto vh-50 list"
  >
    <Component />
  </ul>
</div>
`;
